proto/internal/temporal/server/api/replication/v1/message.proto

```diff
@@ -52,6 +53,7 @@ message ReplicationTask {
         HistoryTaskAttributes history_task_attributes = 8;
         SyncWorkflowStateTaskAttributes sync_workflow_state_task_attributes = 10;
         TaskQueueUserDataAttributes task_queue_user_data_attributes = 11;
+        SyncHSMAttributes sync_hsm_attributes = 14;
```

Comment:
I checked the code path (not in this repo) that uses data blob field but I don't think the impl is ideal from a performance perspective since it stores the mutable state blob in the replication task directly. 
For now since this new task is defined in OSS, I'd keep it the same as the rest of the tasks and use attributes. If we decide to fully migrated to use data blob, we can do it together with all other tasks.